content/en/blog/_posts/2018-08-10-introducing-kubebuilder.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: blog
+title:  'Introducing Kubebuilder: an SDK for building Kubernetes APIs using CRDs'
+date:   2018-08-10
+---
+
+**Author**: Phillip Wittrock (Google), Sunil Arora (Google)
+
+[kubebuilder-repo]: https://github.com/kubernetes-sigs/kubebuilder
+[controller-runtime]: https://github.com/kubernetes-sigs/controller-runtime
+[SIG-APIMachinery]: https://github.com/kubernetes/community/tree/master/sig-api-machinery
+[mailing-list]: https://groups.google.com/forum/#!forum/kubernetes-sig-api-machinery
+[slack-channel]: https://slack.k8s.io/#kubebuilder 
+[kubebuilder-book]: https://book.kubebuilder.io
+[open-an-issue]: https://github.com/kubernetes-sigs/kubebuilder/issues/new
+
+
+How can we enable applications such as MySQL, Spark and Cassandra to manage themselves just like Kubernetes Deployments and Pods do? How do we configure these applications as their own first class APIs instead of a collection of StatefulSets, Services, and ConfigMaps?
+
+We have been working on a solution and are happy to introduce [*kubebuilder*][kubebuilder-repo], a comprehensive development kit for rapidly building and publishing Kubernetes APIs and Controllers using CRDs. Kubebuilder scaffolds projects and API definitions and is built on top of the [controller-runtime][controller-runtime] libraries.
+
+Why Kubebuilder and Kubernetes APIs?  Applications and cluster resources typically require some operational work - whether it is replacing failed replicas with new ones, or scaling replica counts while resharding data.  Running the MySQL application may require scheduling backups, reconfiguring replicas after scaling, setting up failure detection and remediation, etc.
+
+With the Kubernetes API model, management logic is embedded directly into an application specific Kubernetes API, e.g. a “MySQL” API.  Users then declaratively manage the application through yaml configuration using tools such as kubectl, just like they do for Kubernetes objects.  This approach is referred to as an Application Controller, also known as an Operator.  Controllers are a powerful technique backing the core Kubernetes APIs that may be used to build many kinds of solutions in addition to Applications; such as Autoscalers, Workload APIs, Configuration APIs, CI/CD systems, and more.
+
+However, while it has been possible for trailblazers to build new Controllers on top of the raw apimachinery, doing so has been a DIY “from scratch” experience, requiring developers to learn low level details about how Kubernetes libraries are written, handwrite boilerplate code, and warp their own solutions for integration testing, RBAC configuration, documentation, etc.  Kubebuilder makes this experience simple and easy by applying the lessons learned from building the core Kubernetes APIs.
```

Comment:
`apimachinery` -> `API machinery` or `API Machinery`

Reply:
has extra space after `documentation, etc.`

Reply:
`Kubernetes libraries are written, handwrite` -> `Kubernetes libraries are implemented, handwrite`
to avoid written / write in the same sentence.

alternatively split the sentence into two parts.

Reply:
Done.